### ISSUE TITLE
Admin\Operator - External user can select an operator

### ISSUE THREAD

**Issue by BCerki** [bcgov/cas-registration#2047]
#### Dependencies

- Blocked by
- Blocking #1960 

#### Description:

This ticket is about copying the select an operator workflow to reg 2. Wireframes: https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=15413-1900

#### Acceptance Criteria:

Given I am a new user who has never used the app before
When I sign in and navigate to the administration dashboard
Then I see only the select operator and report and issue tiles

Given I click the select operator tile
When am I redirected
Then I see the select an operator dropdown and form (add will be handled in another ticket)

Given I select an operator
When I click "next"
I am asked to confirm I've selected the correct operator

Given I confirm
When I am redirected to a page where I can request access or admin access
Then I can click request

Given I request access
When I am redirected
I can see a confirmation message

#### Development Checklist:

- [x] Refactor dashboard fixtures for conditional display of dashboard links 
- [x] Refactor dashboard tiles for conditional display of dashboard links 
- [x] Refactor middleware for conditional routing 
- [x] Copy the select operator pages into `bciers/apps/administration/app/bceidbusiness/industry_user/select-operator`
- [x] Copy the components in the pages into `bciers/apps/administration/app/components/userOperator`
- [x] Refactor api calls
- [x] Refactor components to work with vitests
- [x] Write missing vitests 
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

Consider fixing this bug as part of this ticket since most of this will be importing a completed feature from reg1. We can add some polish:
https://github.com/bcgov/cas-registration/issues/1428

**Comment by shon-button** [comment]
@patriciarussellCAS

Reachable at:
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/administration/select-operator

### Test Dashboard Tile Link - Select an Operator:
1. Navigate to `http://localhost:3000`
2. Click button "Log in with Business BCeID"
3. Sign in to Keycloak using `bc-cas-dev-secondary`
**Expected results:** 
- Redirection to dashboard page for role industry_user without an associated operator
- Administration tile has link `Select an Operator`
![image](https://github.com/user-attachments/assets/beb2cc7c-c035-4025-83ff-a91793d589ae)
- Link redirects to: 
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/administration/select-operator

**Comment by hannavovk** [comment]
@shon-button The intent is also that a first-time user cannot see any of the other tiles or options/actions until they select an operator and get approved to access that operator. I think the dashboard should show show only the Administration tile and only the 'Select an Operator' action. (Plus the 'report a problem' tile)


![Image](https://github.com/user-attachments/assets/dbaf13b2-de61-4302-a924-004b7037bb07)

I also flagged to @zoeyli-46 that the wireframe currently shows the old Reg 1 tile where the whole tile is for selecting an operator (https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=15682-299&t=JfwXOlGzeqFgIus9-4). I'm assuming that in the current app setup using the Administration tile makes more sense from a dev perspective so suggested a change to the wireframe. Please correct me if it's not the case though.

cc: @patriciarussellCAS

**Comment by patriciarussellCAS** [comment]
The original intent of this ticket appears to be complete. I believe we can close this ticket and address Hanna's comments above in the follow-up ticket #2165 
Devs - let me know if that works for you and I'll close this.

**Comment by patriciarussellCAS** [comment]
Closing this ticket, with a note that we'll need ensure there is a follow-up ticket for greying out the tiles they shouldn't have access to.
